front/components/assistant_builder/AssistantBuilderTablesModal.tsx

```diff
@@ -131,18 +175,22 @@ export default function AssistantBuilderTablesModal({
         tableName: table.name,
       }));
 
-      onSave(configs, selectedDataSource);
+      onSave(configs, selectedDataSourceView);
     } finally {
       setIsSaving(false);
     }
   }
 
-  const onClose = () => {
-    setOpen(false);
-    setTimeout(() => {
-      setSelectedDataSource(null);
-      setIsSaving(false);
-    }, 200);
+  const onClose = (e?: Event, forceClose: boolean = false) => {
+    if (selectedDataSourceView !== null && !forceClose) {
+      setSelectedDataSourceOrView(null);
+    } else {
+      setOpen(false);
+      setTimeout(() => {
```

Comment:
IIRC, this `setTimeout` was needed for the animation 🙈. You can move `Modal` to `ElementModal` so you can remove the `setTimeout`.